src/flask_paranoid/paranoid.py

```
@@ -111,3 +129,51 @@ def clear_session(self, response):
             response.set_cookie(remember_cookie, '', expires=0, max_age=0)
+    
+    def calculate_network_ip(ip_address):
+        """
+        Calculate the network IP by applying the subnet mask to the given IP address.
+
+        This method supports both IPv4 and IPv6 addresses. It calculates the network IP by performing a bitwise AND operation between the IP address and subnet mask.
+
+        Args:
+            ip_address (str): The IP address, either in IPv4 or IPv6 format.
+
+        Returns:
+            str: The network IP address obtained by applying the subnet mask to the IP address.
+
+        Raises:
+            None
+        """
+
+        if ':' in ip_address:
+            subnet_mask = 'ffff:ffff:ffff:ffff::'
+            ip_parts = ip_address.split(':')
+            mask_parts = subnet_mask.split(':')
+
+            ip_binary = ''.join(format(int(part, 16), '016b') for part in ip_parts)
+            mask_binary = ''.join(format(int(part, 16), '016b') for part in mask_parts)
+        else:
+            subnet_mask = '255.255.255.0'
+            ip_parts = ip_address.split('.')
+            mask_parts = subnet_mask.split('.')
+
+            ip_binary = ''.join(format(int(part), '08b') for part in ip_parts)
+            mask_binary = ''.join(format(int(part), '08b') for part in mask_parts)
+
+        network_binary = ''
+        for i in range(len(ip_binary)):
+            network_binary += str(int(ip_binary[i]) & int(mask_binary[i]))
+
+        if ':' in ip_address:
+            network_ip_parts = []
+            for i in range(0, 128, 4):
+                network_ip_parts.append(network_binary[i:i+4])
+            network_ip = ':'.join(network_ip_parts)
+        else:
+            network_ip_parts = []
+            for i in range(0, 32, 8):
+                network_ip_parts.append(str(int(network_binary[i:i+8], 2)))
+            network_ip = '.'.join(network_ip_parts)
+
+        return network_ip
```

The subnet masks for both ipv4 and ipv6 aren't the same for every network, it is fine to provide defaults, but the masks should be configurable.

Storing masks as strings is okay for the configuration, but internally the masks should be stored in a more efficient format that does not require so many conversions for each incoming request.